python/ct/crypto/in_memory_merkle_tree_test.py

```diff
@@ -0,0 +1,193 @@
+    ["leaf", "tree_size_snapshot", "path_length", "path"])
+
+ConsistencyTestVector = namedtuple("ConsistencyTestVector",
+    ["snapshot_1", "snapshot_2", "proof"])
+
+DummySTH = namedtuple("DummySTH", ["tree_size", "sha256_root_hash"])
+
+def decode_hex_strings_list(hex_strings_list):
+    """Decodes a list of hex strings."""
+    return [t.decode("hex") for t in hex_strings_list]
+
+# Leaves of a sample tree of size 8.
+TEST_VECTOR_DATA = decode_hex_strings_list([
+    "",
+    "00",
+    "10",
+    "2021",
+    "3031",
+    "40414243",
+    "5051525354555657",
+    "606162636465666768696a6b6c6d6e6f",
+])
+
+PRECOMPUTED_PATH_TEST_VECTORS = [
+    PathTestVector(0, 0, 0, []),
+    PathTestVector(0, 1, 0, []),
+    PathTestVector(0, 8, 3, decode_hex_strings_list(
+            ["96a296d224f285c67bee93c30f8a309157f0daa35dc5b87e410b78630a09cfc7",
+             "5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e",
+             "6b47aaf29ee3c2af9af889bc1fb9254dabd31177f16232dd6aab035ca39bf6e4"]
+    )),
+    PathTestVector(5, 8, 3, decode_hex_strings_list(
+            ["bc1a0643b12e4d2d7c77918f44e0f4f79a838b6cf9ec5b5c283e1f4d88599e6b",
+             "ca854ea128ed050b41b35ffc1b87b8eb2bde461e9e3b5596ece6b9d5975a0ae0",
+             "d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7"]
+    )),
+    PathTestVector(2, 3, 1, decode_hex_strings_list(
+            ["fac54203e7cc696cf0dfcb42c92a1d9dbaf70ad9e621f4bd8d98662f00e3c125"]
+    )),
+    PathTestVector(1, 5, 3, decode_hex_strings_list(
+            ["6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d",
+             "5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e",
+             "bc1a0643b12e4d2d7c77918f44e0f4f79a838b6cf9ec5b5c283e1f4d88599e6b"]
+    ))]
+
+PRECOMPUTED_PROOF_TEST_VECTORS = [
+    ConsistencyTestVector(1, 1, []),
+    ConsistencyTestVector(1, 8, decode_hex_strings_list(
+            ["96a296d224f285c67bee93c30f8a309157f0daa35dc5b87e410b78630a09cfc7",
+             "5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e",
+             "6b47aaf29ee3c2af9af889bc1fb9254dabd31177f16232dd6aab035ca39bf6e4"]
+    )),
+    ConsistencyTestVector(6, 8, decode_hex_strings_list(
+            ["0ebc5d3437fbe2db158b9f126a1d118e308181031d0a949f8dededebc558ef6a",
+             "ca854ea128ed050b41b35ffc1b87b8eb2bde461e9e3b5596ece6b9d5975a0ae0",
+             "d37ee418976dd95753c1c73862b9398fa2a2cf9b4ff0fdfe8b30cd95209614b7"]
+    )),
+    ConsistencyTestVector(2, 5, decode_hex_strings_list(
+            ["5f083f0a1a33ca076a95279832580db3e0ef4584bdff1f54c8a360f50de3031e",
+             "bc1a0643b12e4d2d7c77918f44e0f4f79a838b6cf9ec5b5c283e1f4d88599e6b"]
+    )),
+    ]
+
+class InMemoryMerkleTreeTest(unittest.TestCase):
+    """Tests for InMemoryMerkleTree."""
+
+    def test_tree_incremental_root_hash(self):
+        """Test root hash calculated correctly when leaves are added
+        incrementally."""
```

Comment:
I think the linter will still complain, as this isn't strictly the format of docstring it allows, but feel free to merge this anyway.